packages/twenty-front/src/modules/object-record/object-filter-dropdown/components/ObjectFilterDropdownSourceSelect.tsx

```diff
@@ -0,0 +1,140 @@
+    objectFilterDropdownSelectedRecordIdsState,
+  );
+  const [fieldId] = useState(v4());
+
+  const selectedFilter = useRecoilValue(selectedFilterState);
+
+  // Dynamically update sourceEnumOptions based on selectedRecordIds
+  const sourceEnumOptions = getSourceEnumOptions(
+    objectFilterDropdownSelectedRecordIds,
+  );
+
+  const filteredSelectedRecords = sourceEnumOptions.filter((option) =>
+    objectFilterDropdownSelectedRecordIds.includes(option.id),
+  );
+
+  const handleMultipleRecordSelectChange = (
+    recordToSelect: SelectableRecord,
+    newSelectedValue: boolean,
+  ) => {
+    const newSelectedRecordIds = newSelectedValue
+      ? [...objectFilterDropdownSelectedRecordIds, recordToSelect.id]
+      : objectFilterDropdownSelectedRecordIds.filter(
+          (id) => id !== recordToSelect.id,
+        );
+
+    if (newSelectedRecordIds.length === 0) {
+      emptyFilterButKeepDefinition();
+      deleteCombinedViewFilter(fieldId);
+      return;
+    }
+
+    setObjectFilterDropdownSelectedRecordIds(newSelectedRecordIds);
+
+    const selectedRecordNames = sourceEnumOptions
+      .filter((option) => newSelectedRecordIds.includes(option.id))
+      .map((option) => option.name);
+
+    const filterDisplayValue =
+      selectedRecordNames.length > MAX_RECORDS_TO_DISPLAY
+        ? `${selectedRecordNames.length} source types`
+        : selectedRecordNames.join(', ');
+
+    if (
+      isDefined(filterDefinitionUsedInDropdown) &&
+      isDefined(selectedOperandInDropdown)
+    ) {
+      const newFilterValue =
+        newSelectedRecordIds.length > 0
+          ? JSON.stringify(newSelectedRecordIds)
+          : EMPTY_FILTER_VALUE;
+
+      const viewFilter =
+        currentViewWithCombinedFiltersAndSorts?.viewFilters.find(
+          (viewFilter) =>
+            viewFilter.fieldMetadataId ===
+            filterDefinitionUsedInDropdown.fieldMetadataId,
+        );
+
+      const filterId = viewFilter?.id ?? fieldId;
+
+      selectFilter({
+        id: selectedFilter?.id ? selectedFilter.id : filterId,
+        definition: filterDefinitionUsedInDropdown,
+        operand: selectedOperandInDropdown || ViewFilterOperand.Is,
+        displayValue: filterDisplayValue,
+        fieldMetadataId: filterDefinitionUsedInDropdown.fieldMetadataId,
+        value: newFilterValue,
+      });
+    }
+  };
+
+  return (
+    <MultipleRecordSelectDropdown
+      selectableListId="object-filter-source-select-id"
+      hotkeyScope={RelationPickerHotkeyScope.RelationPicker}
+      recordsToSelect={sourceEnumOptions.filter(
+        (record) =>
+          !filteredSelectedRecords.some(
+            (selected) => selected.id === record.id,
+          ),
+      )}
+      filteredSelectedRecords={filteredSelectedRecords}
+      selectedRecords={filteredSelectedRecords}
+      onChange={handleMultipleRecordSelectChange}
+      searchFilter={objectFilterDropdownSearchInput}
+      loadingRecords={false}
+    />
+  );
```

Comment:
Here the source is not a record, I think the component `MultipleRecordSelectDropdown` should be generalized in `MultipleSelectDropdown` to allow other types of selection.